### ISSUE TITLE
Implement Delta xDS

### ISSUE THREAD

**Issue by YashMathur** [envoyproxy/java-control-plane#149]
Are there any plans to add an implementation of the delta xDS? I see that there's an [open PR](https://github.com/envoyproxy/go-control-plane/pull/355) in go-control-plane to implement the delta xDS.

**Comment by alecholmez** [comment]
@sschepens I believe you have a fork of this don't you?

**Comment by sschepens** [comment]
Yup, i have this implemented on a fork of mine.
Couple of things:
- Not based on current changes
- Breakes API compatibility (though i think it best)
- Has no tests, but has been running in production for months now

I wont have time to add the correct tests to allow it to be merged, but could probably update it with recent changes in the repo and submit a PR so that someone can continue from that.

**Comment by snowp** [comment]
I don't have a horse in this race but IMO delta is a big enough change that a breaking API (or new API) would be warranted in order to support things such as on demand xDS.

**Comment by sschepens** [comment]
Yeah I think so too, the change I made to the API is not so great, it just basically forces clients to provide each resource with it's corresponding version when building a snapshot.
Will be posting a PR maybe today so that some can get a look at it and maybe someone wants to take it over.

**Comment by slonka** [comment]
@YashMathur do you have time to pick up the work done by @sschepens, add tests / docs and make the PR review ready?
